src/theory/bv/int_blaster.cpp

```diff
@@ -545,6 +545,18 @@ Node IntBlaster::translateWithChildren(
     }
     case kind::APPLY_UF:
     {
+      // The preprocessing pass does not support function applications
+      // with bound variables.
+      for (Node child : original)
+      {
+        if (child.getKind() == Kind::BOUND_VARIABLE)
```

Comment:
Should this be generalized to `expr::hasBoundVar(child)`, e.g. we don't handle any translation where the children of an APPLY_UF are non-ground?

Reply:
hasBoundVar is cheap, it is cached globally per node via an attribute.  

In terms of ensuring we catch all errors, I'm not exactly sure what is allowed of bound variables, are all other cases of bound variables handled on subterms? Can you maybe try an example which replaces the problematic bound variable with a composite term (not under another UF)?

Reply:
About `hasBoundVar`: good to know! thanks.

About sub-terms, you are totally right, of course. I'll fix and add a corresponding regression.

Reply:
fixed and pushed.